pkg/auth/defaultmanager_test.go

```diff
@@ -0,0 +1,51 @@
+package auth_test
+
+import (
+	"log"
+	"os"
+	"testing"
+
+	"github.com/codeready-toolchain/registration-service/pkg/auth"
+	testutils "github.com/codeready-toolchain/registration-service/test"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+type TestDefaultManagerSuite struct {
+	testutils.UnitTestSuite
+}
+
+ func TestRunDefaultManagerSuite(t *testing.T) {
+	suite.Run(t, &TestDefaultManagerSuite{testutils.UnitTestSuite{}})
+}
+
+func (s *TestDefaultManagerSuite) TestKeyManagerDefaultKeyManager() {
+	// Create logger and registry.
+	logger := log.New(os.Stderr, "", 0)
+
+	// Set the config for testing mode, the handler may use this.
+	assert.True(s.T(), s.Config.IsTestingMode(), "testing mode not set correctly to true")
+
+	s.Run("get before init", func() {
+		_, err := auth.DefaultKeyManager()
+		require.Error(s.T(), err)
+		require.Equal(s.T(), "no default KeyManager created, call `InitializeDefaultKeyManager()` first", err.Error())
+	})
+
+	s.Run("first creation", func() {
+		_, err := auth.InitializeDefaultKeyManager(logger, s.Config)
+		require.NoError(s.T(), err)
+	})
+
+	s.Run("second redundant creation", func() {
+		_, err := auth.InitializeDefaultKeyManager(logger, s.Config)
```

Comment:
you should also try to call this `InitializeDefaultKeyManager` with multiple go routines at the same time. See here: https://github.com/fabric8-services/fabric8-auth/blob/b38ee9bf66841ed18af4b01af5dd0ae7f0197b47/worker/worker_blackbox_test.go#L39-L70 In this code, we "prepare" a handful of go routines, but keep them "on hold" and suddenly release them all at once so they call all call the same function in parallel. All but one should fail in your case.

Reply:
Ok, done 5876bb9.